pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -137,6 +137,15 @@ source @out@/nix-support/add-hardening.sh
 extraAfter=($NIX_@infixSalt@_CFLAGS_COMPILE)
 extraBefore=(${hardeningCFlags[@]+"${hardeningCFlags[@]}"})
 
+# When enforcing purity, pretend gcc can't find the current date and
+# time
+if [[ "${NIX_ENFORCE_PURITY:-}" = 1 ]]; then
+    extraAfter=(-D__DATE__=\"???-??-????\"
```

Comment:
Did you mean?

```suggestion
    extraAfter=(-D__DATE__=\"??-??-????\"
```

Reply:
Do you know what debian is doing here?

Reply:
It appears that there has been a patch to gcc (since 2016) to allow using `SOURCE_DATE_EPOCH` https://wiki.debian.org/ReproducibleBuilds/TimestampsFromCPPMacros.

Reply:
Some packages (e.g. pycrypto) do already have `SOURCE_DATE_EPOCH` defined, though it would be preferable to do this globally.

Reply:
There is https://github.com/NixOS/nixpkgs/blob/33b9aa4f35fdc0220987c7f87c1204b5b76f21ad/pkgs/stdenv/generic/setup.sh#L244-L249, but not used by all packages built with gcc.

Reply:
But why we need these macros then, when the environment variable is already exported?

Reply:
If a packages ignores environment variables, we are likely detecting this because cc wrapper will also not find libraries and headers.

Reply:
I suppose since the `SOURCE_DATE_EPOCH` is only supported for gcc >=7.x, but not so for earlier versions, and so I have to modify it to be enabled only for early gcc's.